corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -11,6 +11,37 @@ hqDefine("geospatial/js/gps_capture",[
     initialPageData
 ) {
     'use strict';
+    const selectedMarkerColor = "#00FF00"; // Green
+
+    const MAP_CONTAINER_ID = "geospatial-map";
+    var map;
+    var selectedDataListItem;
+    var coordinateCaptureMarker;
+
+    function setMapVisible(isVisible) {
+        if (isVisible) {
+            $("#" + MAP_CONTAINER_ID).show();
+            centerMapWithMarker();
+        } else {
+            $("#" + MAP_CONTAINER_ID).hide();
+        }
+    }
+
+    function centerMapWithMarker() {
+        if (selectedDataListItem) {
+            let dataItem = selectedDataListItem.itemLocationBeingCapturedOnMap();
+            if (dataItem.lon() && dataItem.lat()) {
+                map.setCenter([dataItem.lon(), dataItem.lat()]);
+                setMarkerAtLngLat(dataItem.lon(), dataItem.lat());
+            }
+        }
+    }
+
+    function runSaveSuccessCallback() {
+        setMapVisible(false);
+        selectedDataListItem.itemLocationBeingCapturedOnMap(null);
+        coordinateCaptureMarker = undefined;
```

Comment:
We should call `coordinateCaptureMarker.remove()` before removing the reference to ensure the old marker is removed from the map. Otherwise, when the user clicks "Save" and tries to capture another point there will be two markers showing on the map.

Reply:
Ah! Good catch!

Reply:
See https://github.com/dimagi/commcare-hq/pull/33489/commits/351fafbf8a0c1d950c1e28d9cfcc3401233509cd